G1ANT.Addon.Selenium/G1ANT.Addon.Selenium/SeleniumAddon.cs

```diff
@@ -40,14 +35,14 @@ public override void LoadDlls()
         private void UnpackDrivers()
         {
             var unpackfolder = AbstractSettingsContainer.Instance.UserDocsAddonFolder.FullName;
-            Dictionary<string, byte[]> exeList = new Dictionary<string, byte[]>()
+            var driversDictionary = new Dictionary<string, byte[]>()
             {
                 { "chromedriver.exe", Resources.chromedriver },
                 { "geckodriver.exe", Resources.geckodriver },
                 { "IEDriverServer.exe", Resources.IEDriverServer },
                 { "MicrosoftWebDriver.exe", Resources.MicrosoftWebDriver }
             };
-            foreach (var exe in exeList)
+            foreach (var exe in driversDictionary.Where(e => !DoesFileExist(unpackfolder, e.Key) || AreFilesOfTheSameLength(e.Value.Length, unpackfolder, e.Key)))
```

Comment:
I think you forgot a exclamation mark before `AreFilesOfTheSameLength(e.Value.Length, unpackfolder, e.Key)`, so the files extracted will be the ones that didn't exist or had DIFFERENT size. I guess the same issue goes to the second PR where you use the same code.

Reply:
And come on, name the variables (exe & driversDictionary) better ot at least comment my remark with the reason why you won't do it. I asked you to do that twice...